src/Microsoft.HttpRepl.IntegrationTests/FileSystem/RealFileSystemTests.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.HttpRepl.FileSystem;
+using Xunit;
+
+namespace Microsoft.HttpRepl.IntegrationTests.FileSystem
```

Comment:
These feel like unit tests more than integration tests, even though they have an concrete dependency that can't be faked.  Thoughts?

Reply:
I went back and forth on that a bit. I would have preferred to make them unit tests, but the reliance on `System.IO.Path.GetTempFileName()` (primarily) and `System.IO.Path.GetTempPath()`  (somewhat) pushed me towards Integration Tests since we wouldn't want to be doing anything with the file system in unit tests.

With one of the changes discussed above, we'd be getting rid of `GetTempFileName()` usage, so that's a positive in the move towards unit tests. But we still have a reliance on `GetTempPath()`. 

So then I guess the question becomes "is it worth it to create a separate `ITempFileSystemProvider` (terrible name, but you get the point) to use here for that one function? 

I'd lean towards no - for the moment at least - in part because it seems reasonable that things dealing with the actual file system (as these particular tests do) are in integration tests.

Thoughts?

Reply:
To me, an integration test is a test that drives the product and focuses on a user scenario rather than the correctness of a single component.  With the removal of `GetTempFileName()` the code being tested here no longer even has side effects on disk; while that's technically an implementation detail, it removes the biggest concern towards low-impact unit tests.  Yes, it does still rely on `Path.GetTempPath()`, but you've already shown that it's easy to test around that, and I don't have concerns about a dependency on a platform-variant behavior since it doesn't have external side effects.

I don't think it requires a further `ITempFileSystemProvider` level of abstraction.  That seems overly complicated at this time.

Reply:
I see what you're getting at. Shifting them to unit tests as-is "just works", so I've done that.